compiler+runtime/src/cpp/jank/runtime/module/loader.cpp

```diff
@@ -954,11 +949,11 @@ namespace jank::runtime::module
       return res;
     }
 
-    loader::set_is_loaded(module);
```

Comment:
What's the reasoning for not having the loader manage this?

Reply:
I assume you're not just asking about _marking a module loaded_ but also _short circuiting if a module is loaded_ in the loader.

This is because if loader manages it, there would be two places that take the same responsibility i.e.
1. Loader itself
2. clojure.core

Major problem is with `:reload` implementation, if we short circuit, module will not be reloaded.

Clojure updates `*loaded-libs*` only in `clojure.core`. We should do the same IMO.